assets/js/modules/reader-revenue-manager/components/setup/SetupMain.js

```diff
@@ -101,8 +101,28 @@ export default function SetupMain( { finishSetup = () => {} } ) {
 	}, [
 		hasResolvedPublications,
 		publicationCreateShown,
-		publications.length,
+		publications,
+		setValues,
+	] );
+
+	const previousPublicationID = usePrevious( publicationID );
+
+	// Do not attempt to reset publication data again once the publication
+	// selection changes.
+	useEffect( () => {
+		if (
+			previousPublicationID !== publicationID &&
+			shouldResetPublications
+		) {
+			setValues( READER_REVENUE_MANAGER_SETUP_FORM, {
+				[ RESET_PUBLICATIONS ]: false,
+			} );
+		}
+	}, [
+		previousPublicationID,
+		publicationID,
 		setValues,
+		shouldResetPublications,
```

Comment:
While working on this issue, I discovered a bug where once a publication reset occurs, it is not triggered again anymore (once you come back to the tab after 15 seconds).

This used to happen because we were setting `RESET_PUBLICATIONS` to `false` unconditionally before running a reset, which is incorrect. It should only be set once the reset is done, AND a publication is selected/changed.

The above change fixes that.

Reply:
Thanks for fixing the issue, @nfmohit 👍